whereby/sdk
# common: Fix dependency versioning process when bumping packages

### Description

Currently, when releasing either the `browser-sdk` or `core` package, we can observe that any `@whereby.com/...` dependency listed in their respective `package.json` file are using a wildcard (`*`) character for the version.

This use of a wildcard version will start to cause backward-compatibility issues in the future since running `yarn install` on an older version of `browser-sdk` or `core` will result in the _very latest_ dependency package being installed. It is likely that any developer running a version of `browser-sdk` <= `2.81` will experience breakage as the underlying core and media packages are further developed in the future.

In this PR we add a changesets link the `browser-sdk`, `core` and `media` packages so a change in one is correctly reflected and propagated throughout all other packages.

### Testing

1. Check out this branch and `yarn install`.
2. Create a new changeset for the `media` package (`yarn changeset`)
3. Apply the new changeset locally with `yarn changeset version`.
4. Looking at the output of the `git diff`, the version of both the `core` package and the `browser-sdk` package should be bumped. In addition, the `@whereby.com/media` dependency in the `core/package.json` and the `@whereby.com/core` dependency in the `browser-sdk/package.json` should point to the latest version.

### Checklist

        under the terms of the MIT license.
